app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,59 @@
+class VendorsController < ApplicationController
+
+  def index
+    @vendors = Vendor.uniq.order(name: :asc)
+  end
+
+  def show
+    @vendors = Vendor.where(id: params[:id])
+  end
+
+  def new
+    @vendor = Vendor.new 
+    if @vendor.market_id
+      @current_market = @vendor.market.id
+    end
+  end
+
+  def create
+    @market = Market.where(id: params[:vendor][:market_id]).first
+    @vendor = Vendor.create(vendor_create_params[:vendor])
+    if @vendor
+      redirect_to market_path(id: @market.id)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @vendor = Vendor.find(params[:id])
+    if @vendor.id
+      @current_market = @vendor.market.id
+    end
+    render :new
+  end
```

Comment:
It will be a good practice to use partials, so the 'edit' and the 'new' will render a _form view file, instead of render the new page through the controller.